src/context.cpp

```diff
@@ -187,19 +188,21 @@ namespace librealsense
         assert( _device_watcher->is_stopped() );
 
 #ifdef BUILD_WITH_DDS
-        if( rsutils::json::get< bool >( settings, "dds-discovery", true ) )
+        if( rsutils::json::get< bool >( settings, std::string( "dds-discovery", 13 ), true ) )
```

Comment:
Do you have to explicitly use the string size? It is inconvenient and error prone

Looks like it should work without the explicit size ([using constructor number 5, not 4](https://en.cppreference.com/w/cpp/string/basic_string/basic_string))

Reply:
It's optional, but I dislike using `strlen` for constants

Reply:
What if you miscount?
Will the json find the right string while searching for `dds-discover` or `dds-discovery ` or would it throw?

Reply:
It's a constant. It's like getting a constant value wrong.
This is why we have a CR and testing :)